src/main/java/apoc/index/IndexUpdateTransactionEventHandler.java

```diff
@@ -0,0 +1,197 @@
+                        }
+                    }
+                });
+    }
+
+    private boolean iterableContains(Iterable<Node> nodes, Node node) {
+        for (Node thisNode: nodes) {
+            if (thisNode.equals(node)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private Collection<Index<Node>> findIndicesAffectedBy(Iterable<Label> labels, String key) {
+        Collection<Index<Node>> result = new ArrayList<>();
+        for (Label label: labels) {
+            Map<String, Collection<Index<Node>>> propertyIndexMap = indexesByLabelAndProperty.get(label.name());
+            if (propertyIndexMap!=null) {
+                final Collection<Index<Node>> indices = propertyIndexMap.get(key);
+                if (indices!=null)  {
+                    result.addAll(indices);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * in async mode add the index action to a collection for consumption in {@link #afterCommit(TransactionData, Collection)}, in sync mode, run it directly
+     */
+    private Void indexUpdate(Collection<Consumer<Void>> state, Consumer<Void> indexAction) {
+        if (state==null) {  // sync
+            indexAction.accept(null);
+        } else { // async
+            state.add(indexAction);
+        }
+        return null;
+    }
+
+    public Map<String, Map<String, Collection<Index<Node>>>> getIndexesByLabelAndProperty() {
+        if (indexesByLabelAndProperty == null ) {
+            initIndexConfiguration();
+        }
+        return indexesByLabelAndProperty;
+    }
+
+    // might be run from a scheduler, so we need to make sure we have a transaction
+    private synchronized void initIndexConfiguration() {
+        try (Transaction tx = graphDatabaseService.beginTx() ) {
+            indexesByLabelAndProperty = new HashMap<>();
+
+            final IndexManager indexManager = graphDatabaseService.index();
+            for (String indexName : indexManager.nodeIndexNames()) {
+                final Index<Node> index = indexManager.forNodes(indexName);
+                Map<String, String> indexConfig = indexManager.getConfiguration(index);
+
+                if (Util.toBoolean(indexConfig.get("autoUpdate"))) {
+                    String labels = indexConfig.getOrDefault("labels", "");
+                    for (String label : labels.split(":")) {
+                        Map<String, Collection<Index<Node>>> propertyKeyToIndexMap = indexesByLabelAndProperty.computeIfAbsent(label, s -> new HashMap<>());
+                        for (String property : indexConfig.getOrDefault("keysForLabel:" + label, "").split(":")) {
+                            final Collection<Index<Node>> indices = propertyKeyToIndexMap.computeIfAbsent(property, s -> new ArrayList<>());
+                            indices.add(index);
+                        }
+                    }
+                }
+            }
+            tx.success();
```

Comment:
I would write back the new hashmap only after tx-success is through, otherwise we might end up with an empty one on error.